src/xunit.performance.api/XunitPerformanceHarness.cs

```diff
@@ -71,6 +75,66 @@ public void RunBenchmarks(string assemblyPath)
             s_profiler(assemblyPath, Configuration.RunId, _outputDirectory, runner, collectOutputFilesCallback);
         }
 
+        public void RunStartupBenchmark(string[] args, ProcessStartInfo psi, PreDel pre, MidDel mid, PostDel post)
+        {
+            pre(_performanceTestConfig, psi);
```

Comment:
OK, let me explain with more detail on why I think this approach is not such a good idea. This is a performance API and not a performance harness, therefore it should only run performance related tasks (stick to the "single responsibility principle" and have a single functionality).

The current design adds an extra functionality to the API with no real value added (the capability to run non-performance-related tasks). Running tasks such as retrieving the repository and building the Jit bench music store benchmarks are not part of this benchmark scenario. Instead, those tasks should be run inside the performance harness, where the user still have control over it.

I believe that the setup and teardown steps should be used to keep the pre/post benchmark steps on a clean state. For example, setup could be used to keep the pre-benchmark step clean and the teardown to collect data. In short, it should be something needed to ensure the benchmark state is clean in every iteration (users want to run these steps for every iteration).

The structure I envision would look more like the pseudo code below:

```
var nIterations = Iterations + 1; // number of iterations specified by the user plus warmup.
for (int i = 0; i < nIterations; ++i)
{
    // setup
    //   start profiling
    //     run benchmark
    //   stop profiling
    // teardown
}
```